Add explanation how to use metadata repository locally (#306)
oracle/graalvm-reachability-metadata: CONTRIBUTING.md

This is the best way to provide support for GraalVM Native Image as it makes an out-of-the-box experience for users (no additional work required) and allows you to continuously test and maintain the metadata as part of your project.
If that is not an option (for example, you are not a maintainer), we encourage you to open a ticket on the issue tracker
of the corresponding library or framework, so that the community can up-vote and discuss the inclusion of reachability metadata
(see an [example](https://github.com/h2database/h2database/issues/3606))

## How to Test or Use This Repository Locally

You can test the reachability metadata from this repository locally against your project or with additional changes.

First, you clone the repository:
```shell
git clone git@github.com:oracle/graalvm-reachability-metadata.git
```
Then, you use the local repository in the configuration of either
[Gradle](https://graalvm.github.io/native-build-tools/latest/gradle-plugin.html#metadata-support) or
[Maven](https://graalvm.github.io/native-build-tools/latest/maven-plugin.html#metadata-support)

## Contribute Metadata

### Checklist

> In order to ensure that all contributions follow the same standards of quality we have devised a following list of requirements for each new added library.

Reviewer: ```suggestion
To ensure that all contributions follow the same standards of quality, we have devised a following list of requirements for each new added library.
```

`org.example:library` project is also included as a template for new libraries.

* GraalVM Reachability Metadata in this repo only contains JSON files as described 
in [Manual Configuration](https://www.graalvm.org/latest/reference-manual/native-image/dynamic-features/Reflection/#manual-configuration)
section of the Native Image documentation.

> * All other library tweaks (such as build time initialization through `native-image.properties`) must not be included
> here. By default, it should be assumed that all user libraries are runtime initialized. Build-time initialization can

Reviewer: ```suggestion
* All other library tweaks (such as build-time initialization through `native-image.properties`) must not be included
here. By default, it should be assumed that all user libraries are run-time initialized. Build-time initialization can
```

# Contributing

> Before contributing to this repository, please try to include the reachability metadata directly into the library. 

Reviewer: ```suggestion
Before contributing to this repository, please consider including reachability metadata directly in the library or the framework.
This is the best way to provide support for GraalVM Native Image as it makes for an out-of-the-box experience for users (no additional work required) and allows you to continuously test and maintain the metadata as part of your project.
```

> If that does not work, open a ticket on the target library issue tracker so the community can upvote and discuss metadata addition.

Reviewer: ```suggestion
If that is not an option (for example, you are not a maintainer), we encourage you to open a ticket on the issue tracker of the corresponding library or framework, so that the community can up-vote and discuss the inclusion of reachability metadata.
```

## How to Test or Use This Repository Locally

You can test the reachability metadata from this repository locally against your project or with additional changes.

First, you clone the repository:
```shell
git clone git@github.com:oracle/graalvm-reachability-metadata.git
```

> Then, you adjust the configuration of local repository in either

Reviewer: ```suggestion
Then, you use the local repository in the configuration of either
```

[Gradle](https://graalvm.github.io/native-build-tools/latest/gradle-plugin.html#metadata-support) or
[Maven](https://graalvm.github.io/native-build-tools/latest/maven-plugin.html#metadata-support)

## Contribute Metadata

### Checklist

> In order to ensure that all contributions follow the same standards of quality we have devised a following list of requirements for each new added library.

Reviewer 2: I still don't see the list here really. Maybe I am missing something.
